usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -207,36 +207,50 @@ def post(self, request):
         """Return all Parents and Recipients matching the provided search text"""
 
         search_text, limit = self.get_request_payload(request)
-
         # Return Recipients with valid id entries
         queryset = LegalEntity.objects.exclude(
             recipient_unique_id__isnull=True).exclude(
             recipient_unique_id__exact='')
 
         # Filter based on search text
         response = {}
+
         # Search and filter for Parent Recipients
         parents = queryset.annotate(
-            search=SearchVector('recipient_name', 'parent_recipient_unique_id')
-        ).filter(
-            search__icontains=search_text,
-            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+            similarity=Greatest(
+                TrigramSimilarity('recipient_name', search_text),
+                TrigramSimilarity('parent_recipient_unique_id', search_text),
+                TrigramSimilarity('recipient_unique_id', search_text))
+        ).distinct().order_by('-similarity').filter(
+            parent_recipient_unique_id__in=F('recipient_unique_id'))
+
         # Search and filter for Recipients, excluding Parent Recipients
         recipients = queryset.annotate(
-            search=SearchVector('recipient_name', 'recipient_unique_id')
-        ).filter(
-            search__icontains=search_text).exclude(
-            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+            similarity=Greatest(
+                TrigramSimilarity('recipient_name', search_text),
+                TrigramSimilarity('recipient_unique_id', search_text),
+                TrigramSimilarity('parent_recipient_unique_id', search_text))
```

Comment:
If we're looking for _just_ recipients here, is the parent DUNS column necessary for the trigram lookup?